flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -575,7 +602,7 @@ protected void resubmitJob(FlinkResourceContext<CR> ctx, boolean requireHaMetada
                 && lastSavepointKnown) {
             specToRecover.getJob().setUpgradeMode(UpgradeMode.SAVEPOINT);
         }
-        restoreJob(ctx, specToRecover, ctx.getObserveConfig(), requireHaMetadata);
+        restoreJob(ctx, specToRecover, ctx.getObserveConfig(), requireHaMetadata, false);
```

Comment:
We could probably leverage the new flag here as well and not always set it to false during resubmit. Maybe when a lost deployment/unhealthy job is restarted it should be true as well. But we can follow up on a separate ticket